ui/src/components/TagSelector/index.tsx

```diff
@@ -329,6 +320,17 @@ const TagSelector: FC<IProps> = ({
         inputRef.current?.focus();
       }
     }
+
+    // set width of tag Form.Control
+    const ele = document.querySelector('.a-input-width') as HTMLElement;
+    if (ele.offsetWidth > 60) {
+      inputRef.current?.setAttribute(
+        'style',
+        `width:${ele.offsetWidth + 16}px`,
+      );
+    } else {
+      inputRef.current?.setAttribute('style', 'width: 60px');
+    }
   }, [focusState, tags, hiddenCreateBtn, searchValue, maxTagLength]);
```

Comment:
The reason for the problem is that the actual width of the `.a-input-width` element cannot be obtained immediately after the value is assigned, so the best solution is to monitor the value of searchValue separately to avoid repeated calculations caused by changes in other values.

```
useEffect(() => {
    const ele = document.querySelector('.a-input-width') as HTMLElement;
    if (ele.offsetWidth > 60) {
      inputRef.current?.setAttribute(
        'style',
        `width:${ele.offsetWidth + 16}px`,
      );
    } else {
      inputRef.current?.setAttribute('style', 'width: 60px');
    }
  }, [searchValue]);

```

By the way, thanks for the contribution, when PRs are added to milestone, we'll work on those PRs before that version is released!